hoodie-client/src/main/java/com/uber/hoodie/table/HoodieCopyOnWriteTable.java

```diff
@@ -379,7 +359,9 @@ public int getPartition(Object key) {
                 List<InsertBucket> targetBuckets = partitionPathToInsertBuckets.get(keyLocation._1().getPartitionPath());
                 // pick the target bucket to use based on the weights.
                 double totalWeight = 0.0;
-                double r = rand.nextDouble();
+                final long totalInserts = Math.max(1, globalStat.getNumInserts());
```

Comment:
should this be `getWorkloadStat(partitionPath).getNumInserts()` and not the global one right.. guess it does not matter?